### PR TITLE
Allow input of Spotify playlist url

### PR DESCRIPTION
**Pull request by wastech**
Allow input of Spotify playlist url yesterday

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -1,46 +1,45 @@
-<!doctype html>
+<!--<!DOCTYPE html>
```
Comment (impactmass) [inline public/index.html:1]: Code duplication. Why did you comment this out and then have similar code below it?

File: server.js [hunk 2]
```diff
@@ -1,63 +1,64 @@
...
 import axios from "axios";
 import cookieParser from "cookie-parser";
 import { generateRandomString } from "./utils";
 
 // TODO: Setup Envalid
-const port = process.env.PORT || 1965;
-const client_id = process.env.CLIENT_ID;
-const client_secret = process.env.CLIENT_SECRET;
-const redirect_uri = process.env.REDIRECT_URI;
+const port = process.env.PORT || 4000;
```
Comment (impactmass) [inline server.js:10]: You don't need to change the port. You have the env var process.env.PORT to set the port.

File: server.js [hunk 3]
```diff
@@ -1,63 +1,64 @@
...
+const client_id = "8293ba9d876042a3895344ac4ba4fbfb";
+const client_secret = "a233161214b448ecb67dc153a127f64d";
```
Comment (impactmass) [inline server.js:12]: Never ever commit secrets into source code. 🔥

This needs to be removed. And should never be committed. Same applies to your client-id. And many other sensitive configurations.

File: server.js [hunk 4]
```diff
@@ -1,63 +1,64 @@
...
+            state: state,
+        })
+    );
 });
 
-app.get("/callback", async (req, res) => {
-  const { state = null, code = null } = req.query;
-  const storedState = req.cookies ? req.cookies[stateKey] : null;
+app.get("/callback", async(req, res) => {
+            const { state = null, code = null } = req.query;
```
[the remark this answers is not in the document]
Comment (impactmass) [reply server.js:42]: Also, I don't think this task needed you to edit all this part of the server.js

File: .env.example [hunk 5]
```diff
@@ -1,4 +1,4 @@
 PORT=1965
-CLIENT_ID=PASTE_YOURS_HERE
+CLIENT_ID="PASTE_YOURS_HERE"
```
[the remark this answers is not in the document]
Comment (wastech) [reply .env.example:2]: alright,

File: package.json [hunk 6]
```diff
@@ -4,7 +4,7 @@
   "description": "tiny",
   "main": "server.js",
   "scripts": {
-    "start": "nodemon --inspect -r esm -r dotenv/config server.js"
+    "start": "nodemon -r esm -r dotenv/config server.js"
```
[the remark this answers is not in the document]
Comment (wastech) [reply package.json]: i have included it

File: package.json [hunk 7]
```diff
@@ -22,7 +22,8 @@
     "cookie-parser": "^1.4.5",
     "esm": "^3.2.25",
     "express": "^4.17.1",
-    "morgan": "^1.10.0"
+    "morgan": "^1.10.0",
+    "request": "^2.88.2"
```
[the remark this answers is not in the document]
Comment (wastech) [reply package.json]: thanks on this ,have removed it

File: public/index.html (hunk 1 above)
Comment (wastech) [reply public/index.html]: i comment it all because of internal css you included

File: server.js (hunk 3 above)
Comment (wastech) [reply server.js]: The proper changes have been made

File: yarn.lock [hunk 8]
```diff
@@ -0,0 +1,2555 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (impactmass) [inline yarn.lock:1]: No yarn.lock. I have started the project with `npm`, and we get package-lock. 

Do not use yarn on this project. We have other projects where yarn is used, but for this one stick to npm. 

So, please remove this file.

**Comment by impactmass** [comment]
This will be done in another form. Closing for now

**Pull request closed without merging**